smartapps/curb/curb-energy-manager.src/curb-energy-manager.groovy

```diff
@@ -0,0 +1,248 @@
+/**
+ *  Curb Smart Energy Max Plus
+ *
+ *  Copyright 2017 Neil Zumwalde
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+
+definition(name: "CURB Energy Manager", namespace: "curb", author: "Neil Zumwalde", description: "Maximize your energy savings with CURB!", category: "", iconUrl: "http://energycurb.com/wp-content/uploads/2015/12/curb-web-logo.png", iconX2Url: "http://energycurb.com/wp-content/uploads/2015/12/curb-web-logo.png", iconX3Url: "http://energycurb.com/wp-content/uploads/2015/12/curb-web-logo.png")
+
+preferences {
+  page(name: "pageOne", nextPage: "pageTwo") {
+    section("Program") {
+      input("name", "text", title: "Program Name", defaultValue: "CURB Energy Manager")
+      input("enabled", "bool", title: "Active", defaultValue: true)
+    }
+    section("When to run") {
+      input("weekdays", "enum", title: "Set Days of Week", multiple: true, required: true,
+              options: ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"],
+                defaultValue: ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday"])
+      input("hours", "enum", title: "Select Times of Day", multiple: true, required: true,
+              options: [[0 : "12am"], [1 : "1am"], [2 : "2am"], [3 : "3am"], [4 : "4am"], [5 : "5am"], [6 : "6am"], [7 : "7am"], [8 : "8am"], [9 : "9am"],
+                [10 : "10am"], [11 : "11am"], [12 : "12pm"], [13 : "1pm"], [14 : "2pm"], [15 : "3pm"], [16 : "4pm"], [17 : "5pm"], [18 : "6pm"], [19 : "7pm"],
+                [20 : "8pm"], [21 : "9pm"], [22 : "10pm"], [23 : "11pm"]],
+                defaultValue: [15, 16, 17, 18, 19])
+    }
+  }
+  page(name: "pageTwo", nextPage: "pageThree" ) {
+    section("Threshold Settings") {
+      input("timeInterval", "enum", title: "Select Measurement Interval", multiple: false,
+              options: [[15 : "15 minutes"], [30 : "30 minutes"], [60 : "60 minutes"]], defaultValue: 30)
+      input("kwhThreshold", "float", title: "Set Threshold Usage (kWh)")
```

Comment:
Consider to use dynamic preference pages with the input option `submitOnChange: true` instead so that you can validate the inputs to make sure they have the right type and are within the valid limits.
http://docs.smartthings.com/en/latest/smartapp-developers-guide/preferences-and-settings.html?highlight=dynamic#dynamic-preferences